### ISSUE TITLE
Sereniry.reportRecordData().fromFile() is not working

### ISSUE THREAD

**Issue by Johan-Sap** [serenity-bdd/serenity-cucumber-starter#132]
Used Serenity BDD Version : **4.0.16** (Also tried upgrading to 4.0.28 but same results)
Java : **19.0.2**
Gradle **8.2**

In some cases we need to add a generated file to the report. Previously we did this using     `Serenity.recordReportData().withTitle("Desktop Screenshot").downloadable().fromFile(Paths.get(imagePath));`

This would add the image in the report at the correct step. Since we upgraded (or at least from 4.0.19 onwards) we see this method is no longer adding the image. Its not even moving it to the 'downloadable' folder in the target/site/serenity/ directory. That folder is not even created.

After debugging a bit myself I found that within the fromFile(..) method at one point an Optional<TestOutCome> is verified to be present.

`Optional<TestOutcome> outcome = eventBus.getBaseStepListener().latestTestOutcome(); if (outcome.isPresent()) {....`

This outcome however remains **Optional.empty** causing the code to never get into the IF statement where (I assume) the actual moving of the file and connecting it to the right step is being done.

The method to add text `Serenity.recordReportData().withTitle("Text").andContents(imagePath);` does work like expected. It seems only files (an image in my case) is not.

**Comment by Johan-Sap** [comment]
Hi, I have a working solution locally. Changed the way its added to follow the method text is added. (Add an Event in the List<StepEventBusEvent> stepEventBusEvents which is used to replay the steps at the end of the test) I tested it in parallel sessions as well and all seems to go as expected. Will be testing a bit more tomorrow just to make sure I haven't overlooked anything.

I did do it without touching other methods in the same classfile, even though I am convinced 2 methods can be refactored to one (the one 1 added and the one that is used for text). But to keep my change separate and minimize impact I decided to leave that for now.
